storage-proxy/nginx.conf

```diff
@@ -0,0 +1,96 @@
+worker_processes  auto;
+
+error_log  /dev/stdout warn;
+pid        /var/run/nginx.pid;
+
+events {
+    worker_connections  1024;
+}
+
+env DOWNLOAD_SPEED;
+env MAX_CONNECTIONS;
+
+http {
+
+    init_by_lua_block {
+        ngx.log(ngx.WARN, "download speed limit: " .. (os.getenv("DOWNLOAD_SPEED") or "not set"))
+        ngx.log(ngx.WARN, "max connections: " .. (os.getenv("MAX_CONNECTIONS") or "not set"))
+    }
+
+    upstream minio_backend {
+        server minio:9000 max_fails=0;
+    }
+
+    lua_shared_dict my_limit_conn_store 100m;
+
+    server {
+
+        proxy_max_temp_file_size 0;
+
+        listen 9000;
+
+        location / {
+            access_by_lua_block {
```

Comment:
Yes. You'd need a custom entrypoint, a config template, accept that GET requests affect other methods. 

This is actually the version I started with, but since there was no reasonable way of just taking values from environment and shoving them into nginx variables I moved to a lua variant.